controllers/linstorsatellite_controller.go

```diff
@@ -530,3 +538,122 @@ func (r *LinstorSatelliteReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			builder.WithPredicates(predicate.Or(predicate.GenerationChangedPredicate{}, predicate.LabelChangedPredicate{}, predicate.AnnotationChangedPredicate{}))).
 		Complete(r)
 }
+
+func PerNodePatches(nodeName string) ([]kusttypes.Patch, error) {
+	nodeAffinityPatch, err := utils.ToEncodedPatch(&kusttypes.Selector{
+		ResId: resid.NewResId(resid.NewGvk("", "v1", "Pod"), "satellite"),
+	}, applycorev1.Pod("satellite", "").
+		WithSpec(applycorev1.PodSpec().
+			WithAffinity(applycorev1.Affinity().
+				WithNodeAffinity(applycorev1.NodeAffinity().
+					WithRequiredDuringSchedulingIgnoredDuringExecution(applycorev1.NodeSelector().
+						WithNodeSelectorTerms(applycorev1.NodeSelectorTerm().
+							WithMatchFields(applycorev1.NodeSelectorRequirement().
+								WithKey("metadata.name").
+								WithOperator(corev1.NodeSelectorOpIn).
+								WithValues(nodeName),
```

Comment:
Is it deliberate that this is now `lsatellite.Name` rather than `node.Name`? Didn't we have the case that the underlying node may change name but the satellite needs to stay on that node with the old name. So we need to match the new node name.

Reply:
I don't think a node in kubernetes can change name. Specifically, in Operator v2, satellite-name == k8s-node-name, always. We can use those interchangeably.